feat: open external sidebar links in new tab

jarrensj opened the pull request:
Add target="_blank" and rel="noopener noreferrer" to all external links in the author sidebar

jarrensj committed:
feat: open external sidebar links in new tab

Add target="_blank" and rel="noopener noreferrer" to all external links in the author sidebar

rjzupkoii commented:
I have mixed feelings about `target="_blank"` since I tend to just open in a new tab by default; however, note that `rel="noopener"` is a bit redundant these days since [modern browsers add it by default](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Attributes/rel/noopener) to `<a>` elements.

jarrensj committed:
fix: standardize external link behavior with target="_blank"

- Add target="_blank" rel="noopener noreferrer" to social media links in footer and share buttons
- Apply same attributes to academic, repository, and documentation links

jarrensj committed:
remove rel="noopener"
